addons/mail/static/src/core/common/notification_message.js

```diff
@@ -0,0 +1,73 @@
+import {
+    Component,
+    onMounted,
+    onPatched,
+    onWillDestroy,
+    onWillUpdateProps,
+    useRef,
+    useState,
+} from "@odoo/owl";
+import { _t } from "@web/core/l10n/translation";
+import { useService } from "@web/core/utils/hooks";
+import { escape } from "@web/core/utils/strings";
+
+export class NotificationMessage extends Component {
+    static template = "mail.NotificationMessage";
+    static props = ["message", "thread", "registerMessageRef?"];
+
+    setup() {
+        super.setup();
+        this.root = useRef("root");
+        onWillUpdateProps((nextProps) => {
+            this.props.registerMessageRef?.(this.props.message, null);
+        });
+        onMounted(() => this.props.registerMessageRef?.(this.props.message, this.root));
+        onPatched(() => this.props.registerMessageRef?.(this.props.message, this.root));
+        onWillDestroy(() => this.props.registerMessageRef?.(this.props.message, null));
+        this.escape = escape;
+        this.store = useState(useService("mail.store"));
```

Comment:
No need to wrap stateful services with `useState()`: `useService()` handled it automatically